src/server/qgsconfigcache.cpp

```diff
@@ -49,9 +49,18 @@ const QgsProject *QgsConfigCache::project( const QString &path, QgsServerSetting
     prj->setBadLayerHandler( badLayerHandler );
 
     QgsProject::ReadFlags readFlags = QgsProject::ReadFlag();
-    if ( ! settings || ! settings->trustLayerMetadata() )
+    if ( settings )
     {
-      readFlags |= QgsProject::ReadFlag::FlagTrustLayerMetadata;
+      // Activate trust layer metadata flag
+      if ( settings->trustLayerMetadata() )
+      {
+        readFlags |= QgsProject::ReadFlag::FlagTrustLayerMetadata;
+      }
+      // Activate don't load layouts flag
+      else if ( settings->getPrintDisabled() )
```

Comment:
I haven't really check, but shouldn't this be a single `if` without the `else` ?

Reply:
You are rigth.